cli-toolchain/graphql.md

```diff
@@ -1739,11 +1754,288 @@ mutation CreateProject {
 }
 ```
 
-> **Note** The **Project.team** resolver is configured to work with the defined connection.
+> **Note** The **Project.team** resolver is configured to work with the defined connection. This is done with a query on the Team table where `teamID` is passed in as the partition key.
 
-Likewise, you can make a simple one-to-many connection as follows:
+Likewise, you can make a simple one-to-many connection as follows for a post that has many comments:
+
+**Has Many**
+
+```graphql
+type Post @model {
+  id: ID!
+  title: String!
+  comments: [Comment] @connection(keyName: "byPost", fields: ["id"])
+}
+
+type Comment @model
+  @key(name: "byPost", fields: ["postID", "content"]) {
+  id: ID!
+  postID: ID!
+  content: String!
+}
+```
+
+Note how a one-to-many connection needs a @key that allows comments to be queried by the postID and the connection uses this key to get all comments whose postID is the id of the post was called on.
+After it's transformed, you can create comments and query the connected Post as follows:
+
+```graphql
+mutation CreatePost {
+  createPost(input: { id: "a-post-id", title: "Post Title" } ) {
+    id
+    title
+  }
+}
+
+mutation CreateCommentOnPost {
+  createComment(input: { id: "a-comment-id", content: "A comment", postID: "a-post-id"}) {
+    id
+    content
+  }
+}
+```
+
+And you can query a Post with its comments as follows:
+
+```graphql
+query getPost {
+  getPost(id: "a-post-id") {
+    id
+    title
+    comments {
+      items {
+        id
+        content
+      }
+    }
+  }
+}
+```
+
+**Belongs To**
+
+You can make a connection bi-directional by adding a many-to-one connection to types that already have a one-to-many connection. In this case we add a connection from Comment to Post since each comment belongs to a post:
+
+```graphql
+type Post @model {
+  id: ID!
+  title: String!
+  comments: [Comment] @connection(keyName: "byPost", fields: ["id"])
+}
+
+type Comment @model
+  @key(name: "byPost", fields: ["postID", "content"]) {
```

Comment:
Is this `@key` needed for this use case? Or is it just an another example of useage?

Reply:
Tested this out, pretty sure for the bi-directional to work we need the @key.